client/ayon_deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -651,6 +574,142 @@ def inject_ayon_environment(deadlinePlugin):
         raise
 
 
+def _wait_for_in_progress(export_url):
+    """Check if another worker started extractenvironments already.
+
+    extractenvironments might be expensive operation, so first worker who
+    starts doing it creates empty file before it. All other workers should
+    wait until this file gets renamed to final name.
+    Raises:
+        (RuntimeError) if extraction takes more
+            than EXTRACT_ENVIRONMENT_TIMEOUT seconds
+    """
+    start_time = datetime.now()
+    while os.path.exists(f"{export_url}.tmp"):
+        date_diff = datetime.now() - start_time
+        if date_diff > timedelta(seconds=EXTRACT_ENVIRONMENT_TIMEOUT):
```

Comment:
It would be nice to be able get the timeout from env variable from job (so we can affect it if necessary).

Reply:
Added check for `AYON_EXTRACT_ENVIRONMENT_TIMEOUT`.